File: app/src/main/kotlin/org/triplepy/sh8email/sh8/activities/login/LoginActivity.java
package org.triplepy.sh8email.sh8.activities.login;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.jakewharton.rxbinding.view.RxMenuItem;
import com.wang.avi.AVLoadingIndicatorView;

import org.triplepy.sh8email.sh8.R;
import org.triplepy.sh8email.sh8.activities.base.BaseActivity;
import org.triplepy.sh8email.sh8.activities.login.di.DaggerLoginComponent;
import org.triplepy.sh8email.sh8.activities.login.di.LoginModule;
import org.triplepy.sh8email.sh8.activities.login.presenter.LoginPresenter;
import org.triplepy.sh8email.sh8.activities.login.presenter.LoginPresenterImpl;
import org.triplepy.sh8email.sh8.activities.mailbox.list.MailListActivity;
import org.triplepy.sh8email.sh8.api.Sh8Client;
import org.triplepy.sh8email.sh8.app.App;

import java.util.concurrent.TimeUnit;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * The sh8email-android Project.
 * ==============================
 * org.triplepy.sh8email.sh8.activities
 * ==============================
 * Created by igangsan on 2016. 9. 3..
 */

public class LoginActivity extends BaseActivity implements LoginPresenter.View {
    @Inject
    public LoginPresenterImpl presenter;

    @BindView(R.id.login_id)
    TextView login_id;
    @BindView(R.id.avi)
    AVLoadingIndicatorView avi;
    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @Inject
    Sh8Client client;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.login_activity);
        DaggerLoginComponent.builder().loginModule(new LoginModule(this)).build().inject(this);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.next_action, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        RxMenuItem.clicks(item)
                .throttleFirst(3, TimeUnit.SECONDS)
                .subscribe(v -> {
                    hideSoftKeyboard();
                    presenter.loginWithId(login_id.getText().toString());
                });
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void navigateToMain(String id) {
        App.setSessionId(id);
        Intent intent = new Intent(this, MailListActivity.class);
        startActivity(intent);
    }

    @Override
    public void showToast(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void showProgressBar() {
        avi.setVisibility(View.VISIBLE);
        avi.show();
    }

    @Override
    public void hideProgressBar() {
        avi.setVisibility(View.GONE);
        avi.hide();
    }

}

File: app/src/main/kotlin/org/triplepy/sh8email/sh8/activities/login/LoginActivity.kt
package org.triplepy.sh8email.sh8.activities.login

import android.app.ActivityOptions
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.transition.TransitionInflater
import android.view.View
import com.jakewharton.rxbinding.view.clicks
import kotlinx.android.synthetic.main.login_activity.*
import org.triplepy.sh8email.sh8.R
import org.triplepy.sh8email.sh8.activities.BaseActivity
import org.triplepy.sh8email.sh8.activities.login.di.DaggerLoginComponent
import org.triplepy.sh8email.sh8.activities.login.di.LoginModule
import org.triplepy.sh8email.sh8.activities.login.presenter.LoginPresenter
import org.triplepy.sh8email.sh8.activities.main.MainActivity
import org.triplepy.sh8email.sh8.ext.toast
import java.util.concurrent.TimeUnit
import javax.inject.Inject

/**
 * The sh8email-android Project.
 * ==============================
 * org.triplepy.sh8email.sh8.activities
 * ==============================
 * Created by igangsan on 2016. 9. 3..
 */

class LoginActivity : BaseActivity(), LoginPresenter.View {
    @Inject
    lateinit var presenter: LoginPresenter

    private val context: Context? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.login_activity)

        DaggerLoginComponent.builder().loginModule(LoginModule(this)).build().inject(this)

        login_nextBtn.clicks()
                .throttleFirst(3, TimeUnit.SECONDS)
                .subscribe {
                    hideSoftKeyboard()
                    presenter.loginWithId(login_id.text.toString())
                }

        setupWindowAnimations()
    }

    private fun setupWindowAnimations() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            val slide = TransitionInflater.from(this).inflateTransition(R.transition.activity_slide)
            window.exitTransition = slide


            /**
             *  hyundee - 화면 넘어 갈때 id가 커지면서 그대로 애니매이션 되도록 하고싶었다.
             *  Enable Window Content Transition - 이것 적용중
             *  인텐트로 화면 전환활때 애니매이션 넣을 view name을 넘기는데
             *  이작업을 여기서 해도 되는가 생각해보자
             * */
            // Text Animation
            val i = Intent(this, MainActivity::class.java)
            val sharedView = login_id
            val transitionName = getString(R.string.trans_login_id)
                    //id.transactionName

            val options = ActivityOptions.makeSceneTransitionAnimation(this, sharedView, transitionName)
            context?.startActivity(intent, options.toBundle())


        }
    }

    override fun showToast(message: String) {
        toast(message)
    }

    override fun showProgressBar() {
        progressBar.visibility = View.VISIBLE
    }

    override fun hideProgressBar() {
        progressBar.visibility = View.GONE
    }
}